pkg/registry/apis/folders/conversions.go

```diff
@@ -26,98 +26,65 @@ func LegacyCreateCommandToUnstructured(cmd *folder.CreateFolderCommand) (*unstru
 			},
 		},
 	}
-	// #TODO: let's see if we need to set the json field to "-"
-	if cmd.UID == "" {
-		cmd.UID = util.GenerateShortUID()
-	}
-	obj.SetName(cmd.UID)
-
-	if err := setParentUID(obj, cmd.ParentUID); err != nil {
-		return &unstructured.Unstructured{}, err
-	}
 
-	return obj, nil
-}
-
-func LegacyUpdateCommandToUnstructured(obj *unstructured.Unstructured, cmd *folder.UpdateFolderCommand) (*unstructured.Unstructured, error) {
```

Comment:
rather than try to do this as a standalone conversion -- I moved this logic to the Update command (the only place where it was called)